airflow/models.py

```diff
@@ -2630,6 +2630,65 @@ def xcom_pull(
             include_prior_dates=include_prior_dates)
 
 
+class ExecutorQueue(Base):
+    __tablename__ = "executor_queue"
+
+    key = Column(String(ID_LEN), primary_key=True)
+    id = Column(String(ID_LEN), nullable=False)
+    updated = Column(DateTime, nullable=False)
+
+
+class ExecutorQueueManager(dict):
+    def __init__(self, *args, **kwargs):
+        super(ExecutorQueueManager, self).__init__(*args, **kwargs)
+        self.session = settings.Session()
+
+    def __getitem__(self, key):
+        qry = (self.session.query(ExecutorQueue)
+               .filter(ExecutorQueue.key == key)
+               .value(ExecutorQueue.id))
+
+        return qry
+
+    def __setitem__(self, key, value):
+        """
+        :param key: task_id
+        :param value: AsyncResult
+        :return:
+        """
+        task = ExecutorQueue()
+        task.key = key
+        task.id = value.id
+        task.updated = datetime.utcnow()
```

Comment:
It might be better to use datetime.now() since everywhere else uses it ( maybe unfortunately).

Reply:
We should actually switch to utcnow() which is the canonical way to do it. We don't use "updated" anywhere at the moment so it suits future compatibility better (as we are saying to use UTC anyway).

Reply:
It's kind of inconsistent in that sense. It might make more sense to migrate everything at once.

Reply:
Agreed. But we are mixing anyway. But a move to utcnow everywhere makes sense. There was a pr out for that but we never accepted it. I might revive that one for 1.9. @criccomini if I do that in the next 2 days would you accept that?

Reply:
@bolkedebruin Ya, if it's merged in by EOW, we can get it into 1.9